### PR TITLE
Use streams

### PR DESCRIPTION
**Pull request by mirpo**
- replace reading and writing to websockets with stream
- replace closing socket with destroying stream, for preventing running not needed events

I don't how it will affect performance, because I use again websocket/ws library. To be tested in broker.

### CODE REVIEW

File: src/connection/WsEndpoint.js [hunk 1]
```diff
@@ -339,7 +310,11 @@ class WsEndpoint extends EventEmitter {
             return
         }
 
-        ws.on('message', (message) => {
+        const duplex = WebSocket.createWebSocketStream(ws, {
```
Comment (harbu) [inline src/connection/WsEndpoint.js:313]: Should we have `websocket-stream` library in package.json?

**Review by harbu — CHANGES_REQUESTED** [review]
LGTM; see one comment about package.json

File: src/connection/WsEndpoint.js (hunk 1 above)
Comment (mirpo) [reply src/connection/WsEndpoint.js:313]: It's not a websocket-stream its a function from ws https://github.com/websockets/ws#use-the-nodejs-streams-api

Comment (harbu) [reply src/connection/WsEndpoint.js:313]: waaat.. nice

**Review by harbu — APPROVED** [review]

**Pull request merged**